packages/cli/src/util/proposerConfig.ts

```diff
@@ -98,3 +104,31 @@ export function readProposerConfigDir(filepath: string, filename: string): Propo
   const proposerConfigJSON = JSON.parse(proposerConfigStr) as ProposerConfigFileSection;
   return proposerConfigJSON;
 }
+
+export function parseBuilderSelection(builderSelection?: string): routes.validator.BuilderSelection | undefined {
+  if (builderSelection) {
+    switch (builderSelection) {
+      case "maxprofit":
+        break;
+      case "builderalways":
+        break;
+      case "builderonly":
+        break;
+      case "executiononly":
+        break;
+      default:
+        throw new YargsError("Invalid input for builder selection, check help");
+    }
+  }
+  return builderSelection as routes.validator.BuilderSelection;
+}
+
+export function parseBuilderBoostFactor(boostFactor?: string): bigint | undefined {
+  if (boostFactor === undefined) return;
+
+  if (!/^\d+$/.test(boostFactor)) {
+    throw new YargsError("Invalid input for builder boost factor, must be a valid number without decimals");
```

Comment:
Just came to my mind that we might wanna use a normal Error here now as this can be thrown from multiple places and not just strictly from the CLI (yargs).

I can address this in another PR, wanted to look into adding `executionalways` but this is currently conflicts with https://github.com/ChainSafe/lodestar/pull/6241, so makes sense to wait on that until that is merged.